lib/main.dart

```diff
@@ -50,53 +50,108 @@ class App extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    // Initialize configuration and service repository.
+    return _WithServiceRepository(
+      child: _WithSelectedNetwork(
+        initialNetwork: initialNetwork,
+        child: MultiBlocProvider(
+          providers: [
+            BlocProvider(
+              create: (context) {
+                // Initialize T&C by loading the currently accepted version from shared preferences.
+                final prefs = context.read<ServiceRepository>().sharedPreferences;
+                return TermsAndConditionAcceptance(prefs);
+              },
+            ),
+          ],
+          child: MaterialApp(
+            routes: appRoutes,
+            theme: concordiumTheme(),
+          ),
+        ),
```

Comment:
```suggestion
        child: BlocProvider(
          create: (context) {
            // Initialize T&C by loading the currently accepted version from shared preferences.
            final prefs = context.read<ServiceRepository>().sharedPreferences;
            return TermsAndConditionAcceptance(prefs);
          },
          child: MaterialApp(
            routes: appRoutes,
            theme: concordiumTheme(),
          ),
        ),
```

Reply:
Considered this change, but it's fairly clear to me that we're going to be providing more blocs here.